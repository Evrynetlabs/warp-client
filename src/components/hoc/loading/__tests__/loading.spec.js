import React from 'react'
import { shallow } from 'enzyme'
import withLoading from 'Components/hoc/loading/loadingContainer'
import configureStore from 'redux-mock-store'

let mockStore

describe('Warp', () => {
  beforeEach(() => {
    mockStore = configureStore()
  })
  describe('When is loading is true', () => {
    it('should render correspondingly', () => {
      const store = mockStore({
        loading: {
          isLoading: true,
        },
      })
      const myComponent = () => {
        return <div></div>
      }
      const WithLoadingComponent = withLoading(myComponent)
      const wrapper = shallow(
        <WithLoadingComponent store={store}></WithLoadingComponent>,
      )
      expect(wrapper).toMatchSnapshot()
    })
  })
  describe('When is stop loading', () => {
    it('should render correspondingly', () => {
      const store = mockStore({
        loading: {
          isLoading: true,
        },
      })
      const myComponent = () => {
        return <div></div>
      }
      const WithLoadingComponent = withLoading(myComponent)
      const wrapper = shallow(
        <WithLoadingComponent store={store}></WithLoadingComponent>,
      )
      expect(wrapper).toMatchSnapshot()
    })
  })
})
