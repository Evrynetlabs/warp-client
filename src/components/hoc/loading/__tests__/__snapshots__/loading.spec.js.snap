// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Warp When is loading is true should render correspondingly 1`] = `
<ContextProvider
  value={null}
>
  <Loading
    isLoading={true}
    startLoading={[Function]}
    stopLoading={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;

exports[`Warp When is stop loading should render correspondingly 1`] = `
<ContextProvider
  value={null}
>
  <Loading
    isLoading={true}
    startLoading={[Function]}
    stopLoading={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
